code/modules/chemistry/tools/food_and_drink.dm

```diff
@@ -1600,3 +1601,25 @@
 	g_amt = 5
 	initial_volume = 40
 	initial_reagents = list("bojack"=40)
+
+/obj/item/reagent_containers/food/drinks/cocktailshaker
+	name = "cocktail shaker"
+	desc = "A stainless steel tumbler with a top, used to mix cocktails. Can hold up to 120 units."
+	icon = 'icons/obj/foodNdrink/bottle.dmi'
+	icon_state = "GannetsCocktailer"
+	initial_volume = 120
+
+	New()
+		..()
+		src.reagents.inert = 1
+
+	attack_self(mob/user)
+		if (src.reagents.total_volume > 0)
+			user.visible_message("<b>[user.name]</b> shakes the container [pick("rapidly", "thoroughly", "carefully")].")
+			playsound(get_turf(src), "sound/items/CocktailShake.ogg", 50, 1, -6)
+			sleep (0.3 SECONDS)
```

Comment:
instead of sleep(), do a time log + comparison 
`
	//add var/shake_time

//condition for checking shake
	if (shake_time > TIME)

//when a shake happens
	shake_time = TIME + 0.3 SECONDS	
`

Reply:
Most of the talk regarding this was on Discord but for now I've decided to leave the code as is. I understand that it's properly not the "proper" way to do it but unless the way I did it is actually nightmaringly awful then I don't feel the motivation to change it to something else.